### PR TITLE
docs: Release notes for 0.14.4.

### PR DESCRIPTION
**Pull request by justin-determined-ai**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/release-notes.txt [hunk 1]
```diff
@@ -8,6 +8,39 @@
...
+**Release Date:** March 17, 2021
+
+**New Features**
+
+-  API: Add a new endpoint exposing historical cluster resource
+   allocation. Currently, information about experiment workloads
+   (training, checkpoints, and validations) is included. Use the
+   :ref:`rest-api` for more information.
+
+-  Introduce a stopping-based variant of
```
Comment (dswang) [inline docs/release-notes.txt:23]: We probably want some categorization of these lines.  i.e. "Scheduler:"

**Review by dswang — APPROVED** [review]

**Review by vishnu2kmohan — APPROVED** [review]
Looks good, but let's wait until we (re)base the K8s coscheduler onto OSS.

File: docs/release-notes.txt (hunk 1 above)
Comment (justin-determined-ai) [reply docs/release-notes.txt:23]: Do we have a master list of categorizations anywhere? I'd prefer not to do an ad-hoc process every time (I actually thought I deleted the API category above, but I think I forgot to save my changes), and I'm not quite sure they actually improve readability or comprehension.

File: docs/release-notes.txt [hunk 2]
```diff
@@ -8,6 +8,39 @@
...
+   :ref:`topic-guides_hp-tuning-det_adaptive-asha` that will continue
+   training trials by default unless stopped by the algorithm. Compared
+   to the default promotions-based algorithm, the stopping variant will
+   promote to higher rungs faster and does not require fault tolerance
+   since it will not resume stopped trials.
+
+-  Add an option to :class:`~determined.pytorch.LRScheduler` to accept a
```
Comment (hamidzr) [inline docs/release-notes.txt:30]: 👍🏼

File: docs/release-notes.txt (hunk 1 above)
Comment (dswang) [reply docs/release-notes.txt:23]: We don't have a master list.  I think they do improve the docs in the sense that people can more easily skip over the notes that they are not interested in.

**Pull request merged**